test/test/CompilerTest.scala

```diff
@@ -234,7 +236,22 @@ abstract class CompilerTest {
       (implicit defaultOptions: List[String]): Unit = {
     val allArgs = args ++ defaultOptions
     val processor = if (allArgs.exists(_.startsWith("#"))) Bench else Main
-    val reporter = processor.process(allArgs)
+    val storeReporter = new Reporter {
+      private val consoleReporter = new ConsoleReporter()
+      private var reporter = new StoreReporter(consoleReporter)
+      def doReport(d: Diagnostic)(implicit ctx: Context): Unit = {
+        if (reporter == null) {
+          consoleReporter.doReport(d)
+        } else {
+          reporter.doReport(d)
```

Comment:
why not wait until test finished and then `if (reporter.hasErrors) reporter.flush`?